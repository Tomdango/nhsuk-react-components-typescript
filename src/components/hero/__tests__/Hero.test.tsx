import React from 'react';
import Hero from '..';
import { shallow, mount } from 'enzyme';

describe('Hero', () => {
  it('matches snapshot', () => {
    const component = shallow(<Hero />);
    expect(component.hasClass('nhsuk-hero--image')).toBeFalsy();
    expect(component).toMatchSnapshot('Hero');
    component.unmount();
  });

  it('adds correct attributes when imageSrc is provided', () => {
    const component = shallow(<Hero imageSrc="image.png"></Hero>);
    expect(component.hasClass('nhsuk-hero--image')).toBeTruthy();
    expect(component.prop('style')).toEqual({ backgroundImage: "url('image.png')" });
    expect(component.find('.nhsuk-hero__overlay').exists()).toBeTruthy();
    component.unmount();
  });

  it('adds correct attributes when imageSrc and children are provided', () => {
    const component = shallow(<Hero imageSrc="image.png">Children</Hero>);
    expect(component.hasClass('nhsuk-hero--image')).toBeTruthy();
    expect(component.hasClass('nhsuk-hero--image-description')).toBeTruthy();
    expect(component.prop('style')).toEqual({ backgroundImage: "url('image.png')" });
    expect(component.find('.nhsuk-hero__overlay').exists()).toBeTruthy();
    component.unmount();
  });

  it('renders HeroContent when there are children', () => {
    const component = mount(<Hero>Children</Hero>);
    expect(component.find('.nhsuk-width-container').exists()).toBeTruthy();
    expect(component.find('.nhsuk-width-container').hasClass('nhsuk-hero--border')).toBeFalsy();
    expect(component.find('.nhsuk-hero-content').exists()).toBeFalsy();
    expect(component.find('.nhsuk-hero__arrow').exists()).toBeFalsy();
    expect(component.find('.nhsuk-hero__wrapper').exists()).toBeTruthy();
    component.unmount();
  });

  it('renders HeroContent when there are children and imageSrc', () => {
    const component = mount(<Hero imageSrc="image.png">Children</Hero>);
    expect(component.find('.nhsuk-width-container').exists()).toBeTruthy();
    expect(component.find('.nhsuk-width-container').hasClass('nhsuk-hero--border')).toBeTruthy();
    expect(component.find('.nhsuk-hero-content').exists()).toBeTruthy();
    expect(component.find('.nhsuk-hero__arrow').exists()).toBeTruthy();
    expect(component.find('.nhsuk-hero__wrapper').exists()).toBeFalsy();
    component.unmount();
  });

  it('HeroContent renders null with no children', () => {
    const component = mount(<Hero></Hero>);
    expect(component.find('.nhsuk-width-container').exists()).toBeFalsy();
    component.unmount();
  });

  describe('Hero.Text', () => {
    it('matches snapshot', () => {
      const component = shallow(<Hero.Text>Text</Hero.Text>);
      expect(component.text()).toBe('Text');
      expect(component).toMatchSnapshot('Hero.Text');
      component.unmount();
    });
  });

  describe('Hero.Heading', () => {
    it('matches snapshot', () => {
      const component = shallow(<Hero.Heading>Text</Hero.Heading>);
      expect(component.text()).toBe('Text');
      expect(component).toMatchSnapshot('Hero.Heading');
      component.unmount();
    });
  });
});
