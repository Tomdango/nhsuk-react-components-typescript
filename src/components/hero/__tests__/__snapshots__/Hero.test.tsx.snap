// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Hero.Heading matches snapshot: Hero.Heading 1`] = `
<h1
  className="nhsuk-u-margin-bottom-3"
>
  Text
</h1>
`;

exports[`Hero Hero.Text matches snapshot: Hero.Text 1`] = `
<p
  className="nhsuk-body-l nhsuk-u-margin-bottom-0"
>
  Text
</p>
`;

exports[`Hero matches snapshot: Hero 1`] = `
<section
  className="nhsuk-hero"
>
  <HeroContent
    hasImage={false}
  />
</section>
`;
